### PR TITLE
API changes to support UI My Registrations table.

### PR DESCRIPTION
**Pull request by doug-lovett**
Signed-off-by: Doug Lovett <doug@diamante.ca>

*Issue #:* /bcgov/entity#7852

*Description of changes:*
- Update get account drafts endpoint response.
- Update get account financing statement endpoint response.
- Add get registrations by account endpoint.
- Add get individual registrations by account endpoints for renewals, discharges, changes, and amendments.
- Update unit tests.
- Spec update pending.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the PPR license (Apache 2.0).

### CODE REVIEW

File: ppr-api/src/ppr_api/models/utils.py [hunk 1]
```diff
@@ -131,6 +130,110 @@
...
+   AND r.registration_type_cl IN ('PPSALIEN', 'MISCLIEN', 'CROWNLIEN')
+   AND r.account_id = '?'
+   AND (fs.expire_date IS NULL OR fs.expire_date > ((now() at time zone 'utc') - interval '30 days'))
+   AND NOT EXISTS (SELECT r3.id
+                     FROM registrations r3
+                    WHERE r3.financing_id = fs.id
+                      AND r3.registration_type_cl = 'DISCHARGE'
+                      AND r3.registration_ts < ((now() at time zone 'utc') - interval '30 days'))
+ORDER BY r.registration_ts DESC
+FETCH FIRST 1000 ROWS ONLY
```
Comment (thorwolpert) [inline ppr-api/src/ppr_api/models/utils.py]: do you want to make these params/envs, so we could alter at runtime without rebuilding the image?

File: ppr-api/src/ppr_api/models/financing_statement.py [hunk 2]
```diff
@@ -297,56 +297,34 @@ def save(self):
...
-        for statement in statement_list:
-            if staff or statement.state_type == model_utils.STATE_ACTIVE:
-                statement_json = {
-                    'matchType': 'EXACT',
-                    'createDateTime': model_utils.format_ts(statement.registration_ts),
-                    'baseRegistrationNumber': statement.registration_num,
-                    'registrationType': statement.registration_type
+
+        query = model_utils.QUERY_ACCOUNT_FINANCING_STATEMENTS.replace('?', account_id)
+        results = db.session.execute(query)
```
Comment (thorwolpert) [inline ppr-api/src/ppr_api/models/financing_statement.py]: I like using binds, for lots of reasons.
```
QUERY_ACCOUNT_FINANCING_STATEMENTS = """
SELECT...
   AND r.account_id = :search_account
...
results = db.session.execute(query, {'search_account': account_id})
```

**Review by thorwolpert — COMMENTED** [review]
Couple of changes to consider, or not:).

File: ppr-api/src/ppr_api/models/financing_statement.py (hunk 2 above)
Comment (doug-lovett) [reply ppr-api/src/ppr_api/models/financing_statement.py]: Change made in second commit.

File: ppr-api/src/ppr_api/models/utils.py (hunk 1 above)
Comment (doug-lovett) [reply ppr-api/src/ppr_api/models/utils.py]: Change made in second commit. Added config variables:

    # DB Query limits on result set sizes
    ACCOUNT_REGISTRATIONS_MAX_RESULTS = os.getenv('ACCOUNT_REGISTRATIONS_MAX_RESULTS', '1000')
    ACCOUNT_DRAFTS_MAX_RESULTS = os.getenv('ACCOUNT_DRAFTS_MAX_RESULTS', '1000')
    ACCOUNT_SEARCH_MAX_RESULTS = os.getenv('ACCOUNT_SEARCH_MAX_RESULTS', '1000')

**Review by thorwolpert — APPROVED** [review]

**Pull request merged**